opentelemetry-instrumentation/src/opentelemetry/instrumentation/genai_utils.py

```diff
@@ -0,0 +1,53 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from os import environ
+from typing import Mapping
+
+from opentelemetry.semconv._incubating.attributes import (
+    gen_ai_attributes as GenAIAttributes,
+)
+from opentelemetry.semconv.attributes import (
+    error_attributes as ErrorAttributes,
+)
+from opentelemetry.trace import Span
+from opentelemetry.trace.status import Status, StatusCode
+from opentelemetry.util.types import AttributeValue
+
+OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT = (
+    "OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT"
+)
+
+
+def is_content_enabled() -> bool:
+    capture_content = environ.get(
+        OTEL_INSTRUMENTATION_GENAI_CAPTURE_MESSAGE_CONTENT, "false"
+    )
+
+    return capture_content.lower() == "true"
+
+
+def get_span_name(span_attributes: Mapping[str, AttributeValue]) -> str:
+    name = span_attributes.get(GenAIAttributes.GEN_AI_OPERATION_NAME, "")
+    model = span_attributes.get(GenAIAttributes.GEN_AI_REQUEST_MODEL, "")
+    return f"{name} {model}"
+
+
+def handle_span_exception(span: Span, error: Exception) -> None:
```

Comment:
maybe

```suggestion
def end_span_with_exception(span: Span, error: Exception) -> None:
```

? 

we're not handling anything, right?

BTW - there is nothing gen-ai specific here, should we move it to common instrumentation helpers?

Reply:
Tbh I'm kind of just mechanically copying from openai-v2: https://github.com/open-telemetry/opentelemetry-python-contrib/blob/a716949d1c28341b2fb9af75f43b541ddeac9335/instrumentation-genai/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/utils.py#L232-L238

Actually, the API/SDK has a mechanism [here when used in `start_as_current_span` ctx manager](https://github.com/open-telemetry/opentelemetry-python/blob/59b225fbf397cdac3b92c3953b6d4b7f29abfb94/opentelemetry-api/src/opentelemetry/trace/__init__.py#L593-L606). That would also call `span.record_exception()` which I believe add a span event with the stacktrace.

cc @lzchen @alizenhom do you know why OpenAI doesn't just use the existing mechanism?